docs/api.md

```diff
@@ -4,211 +4,411 @@ title: single-spa API
+
+<h3>arguments</h3>
+
+<dl class="args-list">
+	<dt>mockWindowLocation: string</dt>
+	<dd>A string representing a window.location that will be used when calling every application's activity function to test if they return true.</dd>
+</dl>
+
+<h3>returns</h3>
+
+<dl class="args-list">
+	<dt>appNames: string[]</dt>
+	<dd>Each string is the name of a registered application that matches the provided <code>mockWindowLocation</code>.</dd>
+</dl>
 
 ## addErrorHandler
-`addErrorHandler(fn)` adds a handler that will be called every time an application throws an error during a lifecycle function or an
-activity function. When there are no error handlers, single-spa throws the error to the window, but when there is at least one handler,
-errors will no longer be thrown on the window.
 
-The `fn` must be a function. It will be called with one argument, `err`, which is an Error object that has a `message` and `name` property.
+Adds a handler that will be called every time an application throws an error during a lifecycle function or activity function. When there are no error handlers, single-spa throws the error to the window.
+
+<dl class="args-list">
+	<dt>errorHandler: Function(error: Error)</dt>
+	<dd>Must be a function. Will be called with an <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error">Error object</a> that additionally has a <code>message</code> and <code>name</code> property.</dd>
+</dl>
+
+<h3>returns</h3>
+
+`undefined`
 
 ## removeErrorHandler
-`removeErrorHandler(fn)` removes an error handler. Returns true if the error handler was removed, and false if it was not.
+
+Removes the given error handler function. 
+
+<h3>arguments</h3>
+
+<dl class="args-list">
+	<dt>errorHandler: Function</dt>
+	<dd>Reference to the error handling function.</dd>
+</dl>
+
+<h3>returns</h3>
+
+<dl class="args-list">
+	<dt>boolean</dt>
+	<dd><code>true</code> if the error handler was removed, and <code>false</code> if it was not.</dd>
+</dl>
 
 ## mountRootParcel
-`mountRootParcel(parcelConfig, parcelProps)` will create and mount a parcel.
-This function takes a [parcelConfig](parcels-api.md#parcel-configuration) and [parcelProps](parcels-api.md#parcel-props).
-The mounted parcel will not be automatically unmounted and unmounting will need to be triggered manually.
 
-## before routing event
-single-spa fires an event `single-spa:before-routing-event` on the window every time before a routing event occurs.
-This event will get fired after each hashchange, popstate, or triggerAppChange, even if no changes
-to registered applications were necessary. Sample usage of this event might look like this:
+Will create and mount a [single-spa parcel](parcels-overview.md).
+
+> Note: parcels do not automatically unmount. Unmounting will need to be triggered manually.
+
+<h3>arguments</h3>
+
+<dl class="args-list">
+	<dt>parcelConfig: ???</dt>
```

Comment:
Oops forgot to remove this. I plan on submitting updated parcels docs in a future PR.